utils/compare/graph.py

```diff
@@ -0,0 +1,306 @@
+
+    return all_with_result
+
+def process_all_data(metric_of_interest, find_metrics,uuid, ids, data_func, index, divider, additional_columns): 
+    # Process all
+
+    columns = []
+    for k in find_metrics.keys(): 
+        columns.append(k)
+    for column_name in additional_columns: 
+        columns.append(column_name)
+    columns.append(metric_of_interest)
+    # Get all previous run data
+    oData = getResults(uuid,ids,index, find_metrics)
+    oMetrics = processData(oData, columns, data_func)
+    
+    oMetrics = oMetrics.reset_index()
+    oMetrics[metric_of_interest] = oMetrics[metric_of_interest] /divider
+
+    #Current data metrics 
+    cData = getResults(uuid,[uuid],index, find_metrics)
+    nMetrics = processData(cData, columns, data_func)
+    nMetrics = nMetrics.reset_index()
+    nMetrics[metric_of_interest] = nMetrics[metric_of_interest] /divider
+    return oMetrics, nMetrics, columns
+
+def generate_pass_fail(oMetrics, nMetrics, columns, tolerancy_num):
+
+    all_merged = pd_merge(oMetrics, nMetrics, columns)
+    print('all merged' + str(all_merged))
+    if not all_merged.empty:
+        all_result_tolerancy = tolerancy(all_merged, tolerancy_num)
+        return all_result_tolerancy
+    return []
+
+def jobSummary(uuids: list):
+    index = "ripsaw-kube-burner*"
+    ids = "\" OR uuid: \"".join(uuids)
+    query = {
+        "query": {
+            "query_string": {
+                "query": (
+                    f'( uuid: \"{ids}\" )'
+                    f' AND metricName: "jobSummary"'
+                    )
+            }
+        }
+    }
+    #print(query)
+    es = ElasticService(index=index)
+    response = es.post(query)
+    es.close()
+    runs = [item['_source'] for item in response["hits"]["hits"]]
+    return runs
+
+def processData(data: dict, columns: list[str], func="mean") :
+    #pprint.pprint(data)
+    df = pd.json_normalize(data)
+    filterDF = burnerFilter(df, columns)
+    if func == "avg" or func == "mean": 
+        ptile = filterDF.groupby(columns[:-1])[columns[-1]].mean()
+    elif func == "max": 
+        ptile = filterDF.groupby(columns[:-1])[columns[-1]].max()
+    return ptile
+
+def processBurner(data: dict) :
+    #pprint.pprint(data)
+    df = pd.a(data)
+    filterDF = burnerFilter(df)
+    #print('filter df ' + str(filterDF))
+    ptile = filterDF.groupby(['quantileName'])['P99'].mean()
+    #print("p tile" + str(ptile))
+    return ptile
+
+def jobFilter(pdata: dict, data: dict):
+    columns = ['uuid','jobConfig.jobIterations']
+    pdf = pd.json_normalize(pdata)
+    pick_df = pd.DataFrame(pdf, columns=columns)
+    iterations = pick_df.iloc[0]['jobConfig.jobIterations']
+    df = pd.json_normalize(data)
+    ndf = pd.DataFrame(df, columns=columns)
+    ids_df = ndf.loc[df['jobConfig.jobIterations'] == iterations ]
+    return ids_df['uuid'].to_list()
+
+def burnerFilter(data: dict, columns: list[str]) :
+    #
```

Comment:
If interested in keeping doc strings, we can keep the format same across: https://google.github.io/styleguide/pyguide.html